lib/common/models/redis_store.rb

```diff
@@ -53,6 +53,10 @@ def self.find(redis_key = nil)
       end
     end
 
+    def self.find_or_create(redis_key)
+      find(redis_key) || new(Hash[@redis_namespace_key, redis_key])
```

Comment:
Note: If you want to just change the name of the method to `find_or_build` and client/session will effectively be overriding that, this could work for now and I can DRY it up in a follow-up PR later.